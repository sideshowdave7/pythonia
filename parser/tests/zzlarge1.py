global a

def useless_init(arg):
 pass

def moar_func():
 x = []
 for i in range(10):
  x.append(lambda x, y: i*-x+y)

def main():
 useless_init("a" "b")
 def iner_main(arg1, arg2):
  return arg1[arg2 >> (3 + 4)]
 ident = lambda x: x
 b = lambda: 3
 ident(iner_main([1, 2, 3, 4], 12))()
 x = moar_func()


if __name__ == '__main__'():
 main()

