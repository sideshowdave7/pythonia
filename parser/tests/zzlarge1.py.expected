(program
 (global a)
 (def (useless_init arg) (suite (pass)))
 (def
  (moar_func)
  (suite
   (= (x) (list))
   (for
    i
    (indexed range (called 10))
    (suite
     (expr
      (indexed
       x
       (dot append)
       (called (lambda (x y) (arith (term i ("*" ("-" x))) ("+" y))))))))))
 (def
  (main)
  (suite
   (expr (indexed useless_init (called "ab")))
   (def
    (iner_main arg1 arg2)
    (suite
     (return
      (indexed arg1 (subscript (shift arg2 (">>" (arith 3 ("+" 4)))))))))
   (= (ident) (lambda (x) x))
   (= (b) (lambda () 3))
   (expr
    (indexed
     ident
     (called (indexed iner_main (called (list 1 2 3 4) 12)))
     (called)))
   (= (x) (indexed moar_func (called)))))
 (cond
  ((comparison __name__ ("==" (indexed "__main__" (called))))
   (suite (expr (indexed main (called)))))))

